h2o-core/src/main/java/water/init/NodePersistentStorage.java

```diff
@@ -123,7 +123,7 @@ public void put(String categoryName, String keyName, InputStream is) {
         throw new RuntimeException("Could not make NodePersistentStorage category directory (" + tmpd + ")");
       }
     }
-    if (! pm.exists(tmpd)) {
+    if (! H2O.getPM().isGcsPath(tmpd) && ! pm.exists(tmpd)) {
```

Comment:
I wouldn't introduce this special case handling for GCS. Frankly, I would rather remove this check completely. I don't see the purpose of checking for the existence of the directory right after we explicitly created it.

Suggestion: remove lines 126-128. Your thoughts @Pscheidl ?

Similar on lines 138-140